Source/engine.cpp

```diff
@@ -555,6 +555,32 @@ void DrawLineTo(CelOutputBuffer out, int x0, int y0, int x1, int y1, BYTE color_
 	}
 }
 
+void DrawHalfTransparentRectTo(CelOutputBuffer out, int sx, int sy, int width, int height)
+{
+	int row, col;
+	BYTE *pix = out.at(sx, sy);
+
+	if (sgOptions.bBlendedTransparancy) { // Blended

```

Comment:
maybe we should split this in to two function?

Reply:
Done